subsys/bluetooth/controller/hci_driver.c

```diff
@@ -621,14 +621,16 @@ static int configure_memory_usage(void)
 		return required_memory;
 	}
 
-	cfg.adv_count.count = SDC_ADV_SET_COUNT;
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
```

Comment:
We need to be able to set the number of advertising sets when not supporting the peripheral
```suggestion
	if (IS_ENABLED(CONFIG_BT_BROADCASTER)) {
```